cylc/flow/cfgspec/suite.py

```diff
@@ -351,27 +351,26 @@
             supports use of the 365 (never a leap year) and 366 (always a leap
             year) calendars.
         ''')
-        Conf('runahead limit', VDR.V_STRING, desc='''
+        Conf('runahead limit', VDR.V_STRING, 'P5', desc='''
             Runahead limiting prevents the fastest tasks in a suite from
             getting too far ahead of the slowest ones, as documented in
             :ref:`RunaheadLimit`.
 
-            This config item specifies a hard limit as a cycle interval
-            between the slowest and fastest tasks. See also
-            :cylc:conf:`[..]max active cycle points` which defines the limit
-            as a number of cyles.
-
-            Example: ``PT12H`` - for a 12 hour limit under ISO 8601 cycling.
-        ''')
-        Conf('max active cycle points', VDR.V_INTEGER, 3, desc='''
-            Runahead limiting prevents the fastest tasks in a suite from
-            getting too far ahead of the slowest ones, as documented in
-            :ref:`RunaheadLimit`.
-
-            It allows up to ``N`` (default 3)
-            consecutive cycle points to be active at any time, adjusted up if
-            necessary for any future triggering.
-        ''')
+            This limit on the number of consecutive cycle points is specified
+            by an interval between the least and most recent: either an integer
+            interval (e.g. ``P3`` -  works for both :term:`integer cycling` and
+            :term:`datetime cycling`), or a time interval (e.g. ``PT12H`` -
+            only works for datetime cycling). Alternatively, if a raw number is
+            given, it will be taken to mean that number of hours, though this
+            usage is deprecated.
+        ''')  # TODO: mention :cylc:conf:`[..]max active cycle points`
+        Conf('max active cycle points', VDR.V_INTEGER, None, desc='''
+        ''')  # TODO: rewrite desc after non-consecutive implementation
```

Comment:
What do we want to do with `max active cycle points`? Options:

1) Deprecate it and upgrade value to `runahead limit`. When we get around to the non-consecutive limit we would have to use a different configuration name or move it to a new section.

2) Keep it and upgrade existing non-consecutive limits to consecutive ones.

Reply:
Thinking about it a bit I think we have multiple suites built to the same pattern which are dependent on consecutive behaviour which would constrain us to (1).

Reply:
You're probably right. Unfortunately, because "max active cycle points" is the obvious best name for the new item.  I guess we could go for "maximum active cycle points" :grin:

Reply:
https://github.com/cylc/cylc-flow/issues/3667#issuecomment-705183728